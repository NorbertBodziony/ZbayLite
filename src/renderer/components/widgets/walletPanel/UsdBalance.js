import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

const styles = {
  value: {
    fontSize: '20px'
  },
  title: {
    fontSize: '0.71rem'
  }
}

const UsdBalance = ({ classes, value }) => (
  <Grid container direction='column'>
    <Typography variant='body2' className={classes.title}>
      Available
    </Typography>
    <Typography variant='h5' className={classes.value}>
      ${R.isNil(value) ? '-' : value} USD
    </Typography>
  </Grid>
)

UsdBalance.propTypes = {
  classes: PropTypes.object.isRequired,
  value: PropTypes.string
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(UsdBalance)
