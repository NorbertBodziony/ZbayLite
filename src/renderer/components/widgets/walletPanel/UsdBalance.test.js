/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'
import BigNumber from 'bignumber.js'

import { mockClasses } from '../../../../shared/testing/mocks'
import { UsdBalance } from './UsdBalance'

describe('UsdBalance', () => {
  it('renders component', () => {
    const result = shallow(
      <UsdBalance classes={mockClasses} value={new BigNumber('3243.25')} />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders when no value', () => {
    const result = shallow(
      <UsdBalance classes={mockClasses} />
    )
    expect(result).toMatchSnapshot()
  })
})
