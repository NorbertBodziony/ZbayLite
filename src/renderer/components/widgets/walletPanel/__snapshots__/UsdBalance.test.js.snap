// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsdBalance renders component 1`] = `
<WithStyles(Typography)
  className="value-mock"
  variant="h5"
>
  $ 
  3,243.25
   USD
</WithStyles(Typography)>
`;

exports[`UsdBalance renders when no value 1`] = `
<WithStyles(Typography)
  className="value-mock"
  variant="h5"
>
  $ 
  -
   USD
</WithStyles(Typography)>
`;
